pkg/autoscaler/metrics/stats_scraper.go

```diff
@@ -117,7 +118,8 @@ var noKeepaliveClient = &http.Client{
 // to take advantage of HTTP Keep-Alive to avoid connection creation overhead
 // between scrapes of the same pod.
 var client = &http.Client{
-	Timeout: httpClientTimeout,
+	Timeout:   httpClientTimeout,
+	Transport: pkgnet.NewAutoTransport(),
```

Comment:
bumping `MaxIdleConns` seems pretty reasonable to me here, but do we want to pull in the whole NewAutoTransport with HTTP2 support, or just directly include a `&http.Transport{ MaxIdleConns: 1000 }` here (similar to how we do it on line 110 above for noKeepaliveClient)?

Reply:
I assume we should also specify the IdleConnTimeout for transport as the default value 0 means no timeout.  I did this change as there are already a bespoke transport in NewAutoTransport. But I have no idea of the impact for HTTP2 support. 
I think it is reasonable to just specify the parameters for transport we need. If so we can also leave the MaxIdleConnsPerHost field to default value of 2.

Reply:
yeah, in practice I don't think we'd ever have more than one connection open to a single host since we'd never be scraping the same QP in parallel, if you see what I mean? I'd argue we already use defaults for the other transport fields in noKeepaliveClient so I assume they're ok as-is.

Reply:
So I will change the transport construction accordingly.
I think we cannot omit the IdleConnTimeout as the same of noKeepaliveClient.  What do you think?

Reply:
ah yeah, I see what you mean about IdleConnTimeout since that doesn't apply to noKeepaliveClient as there's never a keep-alive conn to be idle. Yeah, seems like it makes sense to move that over and set it explicitly.

Reply:
While this works, there's no guarantee that default transport won't change the values. 
So if we want specific values here — we should just create a specific transport here.

Reply:
I have changed the commit to set MaxIdleConns and IdleConnTimeout explicitly. Thanks!